mlflow/models/model.py

```diff
@@ -124,15 +142,8 @@ def log(cls, artifact_path, flavor, registered_model_name=None,
             local_path = tmp.path("model")
             run_id = mlflow.tracking.fluent._get_or_start_run().info.run_id
             mlflow_model = cls(artifact_path=artifact_path, run_id=run_id)
-            if signature is not None:
```

Comment:
To make sure I understand - before, we could only save signature/examples via `log_model`, which saved them here (the operation of saving a signature/example is flavor-agnostic). Now, we've enabled saving signature/examples in `save_model`, which doesn't have a common code-path across flavors (i.e. each flavor defines its own `save_model` implementation), hence why we duplicate the code for saving signature/examples across the flavors.

Reply:
yes, exactly.